src/components/pdfviewer/PdfViewer.jsx

```diff
@@ -0,0 +1,42 @@
+import { useState, useEffect } from "react";
+import { pdfBase64Example } from "./pdfbase64example";
+
+const PdfViewer = ({
+  filePdf = pdfBase64Example,
+  width = "100%",
+  height = "700px",
+}) => {
+  const [createdLinks, setCreatedLinks] = useState("");
+
+  useEffect(() => {
+    const convertBase64ToUrlLink = (pdfdecoded) => {
+      if (filePdf) {
+        const removeBase64DataType = pdfdecoded.split(",")[1];
+        const byteCharacters = atob(removeBase64DataType);
+        const byteNumbers = new Array(byteCharacters.length);
+        for (let i = 0; i < byteCharacters.length; i++) {
+          byteNumbers[i] = byteCharacters.charCodeAt(i);
+        }
+        const byteArray = new Uint8Array(byteNumbers);
+        const file = new Blob([byteArray], { type: "application/pdf" });
+        const fileURL = URL.createObjectURL(file);
+        setCreatedLinks(fileURL);
+        return fileURL;
+      } else {
+        return "";
+      }
+    };
+
+    convertBase64ToUrlLink(filePdf);
+
+    if (createdLinks !== "") {
+      URL.revokeObjectURL(createdLinks);
+    }
+  }, [filePdf]);
```

Comment:
Have a look at https://react.dev/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes . Try out the other approach without a useEffect

Reply:
I think I need useEffect because I want to delete the link when the component is closed